dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -368,6 +375,23 @@ private List<Event> inlineAssert(FunctionCall call) {
         return List.of(assertion, abort);
     }
 
+    private List<Event> integerOverflow(FunctionCall call) {
+        ExpressionFactory expressions = ExpressionFactory.getInstance();
+        final Expression condition = expressions.makeFalse();
+        final Event assertion = EventFactory.newAssert(condition, "integer overflow");
+        final Event abort = EventFactory.newAbortIf(expressions.makeTrue());
+        abort.addTags(Tag.EARLYTERMINATION, Tag.OVERFLOW);
+        return List.of(assertion, abort);
```

Comment:
You are adding the tag to the abort, not to the assertion. I don't think that is intended.

Also, is the only purpose of the tag to distinguish this from user assertions so that the user can disable normal assertions while still having these?
That makes `Tag.NULLDEREF` and `Tag.OVERFLOW` functionally identical (for now).
Wouldn't it be easier and more general to allow the user to specify a set of strings to disable any property he wants?
Something like `properties.disable=user_assertion, invalid_deref, overflow`.